@TrustHenry help me I have a Windows linker error (your favorite)!

```
agora-unittests.obj : error LNK2019: unresolved external symbol "public: static class std::shared_ptr<class stellar::QuorumIntersectionChecker *> __cdecl stellar::QuorumIntersectionChecker::create(struct std::unordered_map<struct stellar::PublicKey,class std::shared_ptr<struct stellar::SCPQuorumSet>,struct std::hash<struct stellar::PublicKey>,struct std::equal_to<struct stellar::PublicKey>,class std::allocator<struct std::pair<struct stellar::PublicKey const ,class std::shared_ptr<struct stellar::SCPQuorumSet> > > > const &)" (?create@QuorumIntersectionChecker@stellar@@SA?AV?$shared_ptr@PEAVQuorumIntersectionChecker@stellar@@@std@@AEBU?$unordered_map@UPublicKey@stellar@@V?$shared_ptr@USCPQuorumSet@stellar@@@std@@U?$hash@UPublicKey@stellar@@@4@U?$equal_to@UPublicKey@stellar@@@4@V?$allocator@U?$pair@$$CBUPublicKey@stellar@@V?$shared_ptr@USCPQuorumSet@stellar@@@std@@@std@@@4@@4@@Z) referenced in function _D5agora9consensus6Quorum22verifyQuorumsIntersectFHSQBy6common6crypto3Key9PublicKeySQDeQBg5Types12QuorumConfigZv
```

--- a/source/agora/consensus/EnrollmentManager.d
+++ b/source/agora/consensus/EnrollmentManager.d
@@ -522,6 +522,26 @@ public class EnrollmentManager
         return PublicKey(this.key_pair.V[]);
     }
 
+    /// Returns: true if this validator is currently enrolled
+    public bool isEnrolled (UTXOFinder finder) nothrow @safe
+    {
+        Hash[] utxo_keys;
+        assert(this.validator_set.getEnrolledUTXOs(utxo_keys));
+
+        const PublicKey key = this.getEnrollmentPublicKey();
+        foreach (utxo_key; utxo_keys)
+        {
+            UTXOSetValue value;
+            if (!finder(utxo_key, size_t.max, value))
+                assert(0, "UTXO for validator not found!");  // should never happen
+
+            if (value.output.address == key)
+                return true;
+        }
+
+        return false;
+    }
+
     /***************************************************************************
 
         Get the key for the enrollment data for this node

--- a/source/agora/consensus/Quorum.d
+++ b/source/agora/consensus/Quorum.d
@@ -77,35 +77,6 @@ unittest
 
 /*******************************************************************************
 
-    Verify that the provided quorum configuration is considered sane by SCP.
-
-    Params:
-        quorum = the quorum config
-
-    Throws:
-        an Exception if the quorum is not considered sane by SCP.
-
-*******************************************************************************/
-
-private void verifyQuorumSanity (const ref QuorumConfig quorum)
-{
-    import scpd.scp.QuorumSetUtils;
-    auto scp_quorum = toSCPQuorumSet(quorum);
-
-    const bool ExtraChecks = true;
-    const(char)* fail_reason;
-    enforce(isQuorumSetSane(scp_quorum, ExtraChecks, &fail_reason),
-        format("Quorum %s fails sanity check before normalization: %s",
-                quorum, fail_reason.fromStringz));
-
-    normalizeQSet(scp_quorum);
-    enforce(isQuorumSetSane(scp_quorum, ExtraChecks, &fail_reason),
-        format("Quorum %s fails sanity check after normalization: %s",
-                quorum, fail_reason.fromStringz));
-}
-
-/*******************************************************************************
-
     For each enrollment find the associated public key,
     and return the set of keys.
 
@@ -189,4 +160,34 @@ private auto genTestEnrolls (size_t num_enrollments)
     PublicKey[] keys = key_pairs.map!(k => cast()k.address).array;
     keys.sort();
     return Result(keys, enrolls, &storage.findUTXO);
+}
+
+/*******************************************************************************
+
+    Verify that the provided quorum configuration is considered sane by SCP.
+
+    Params:
+        quorum = the quorum config
+
+    Throws:
+        an Exception if the quorum is not considered sane by SCP.
+
+*******************************************************************************/
+
+version (unittest)
+private void verifyQuorumSanity (const ref QuorumConfig quorum)
+{
+    import scpd.scp.QuorumSetUtils;
+    auto scp_quorum = toSCPQuorumSet(quorum);
+
+    const bool ExtraChecks = true;
+    const(char)* fail_reason;
+    enforce(isQuorumSetSane(scp_quorum, ExtraChecks, &fail_reason),
+        format("Quorum %s fails sanity check before normalization: %s",
+                quorum, fail_reason.fromStringz));
+
+    normalizeQSet(scp_quorum);
+    enforce(isQuorumSetSane(scp_quorum, ExtraChecks, &fail_reason),
+        format("Quorum %s fails sanity check after normalization: %s",
+                quorum, fail_reason.fromStringz));
 }

--- a/source/agora/node/Validator.d
+++ b/source/agora/node/Validator.d
@@ -91,6 +91,9 @@ public class Validator : FullNode, API
     {
         try
         {
+            if (!this.enroll_man.isEnrolled(this.utxo_set.getUTXOFinder()))
+                return;
+
             this.qc = this.rebuildQuorumConfig();
             this.nominator.setQuorumConfig(this.qc);
             buildRequiredKeys(this.config.node.key_pair.address, this.qc,
@@ -212,6 +215,9 @@ public class Validator : FullNode, API
 
     protected final override void onAcceptedTransaction () @safe
     {
+        if (!this.enroll_man.isEnrolled(this.utxo_set.getUTXOFinder()))
+            return;  // nothing to do, we're not an active validator
+
         // check if there's enough txs in the pool, and start nominating
         this.nominator.tryNominate();
     }

